clients/src/main/java/org/apache/kafka/common/utils/Checksums.java

```diff
@@ -30,21 +36,74 @@
  */
 public final class Checksums {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(Checksums.class);
+    private static final MethodHandle BYTE_BUFFER_UPDATE;
+
+    static {
+        MethodHandle byteBufferUpdate = null;
+        if (Java.IS_JAVA9_COMPATIBLE) {
+            try {
+                byteBufferUpdate = MethodHandles.publicLookup().findVirtual(Checksum.class, "update", MethodType.methodType(void.class, ByteBuffer.class));
+            } catch (Throwable t) {
+                LOGGER.warn("Cannot uses java.util.zip.Checksum::update(ByteBuffer): direct ByteBuffers checksum won't use optimized platform support", t);
```

Comment:
Linking the related resolved comment on https://github.com/apache/kafka/pull/12163#discussion_r873652472

Reply:
Thanks. I am still not clear why we'd rather warn than fail. With the current code, if there's a bug in this logic, our tests won't catch it (we'll simply have a logged warning).

Reply:
Yep, it means a that you won't get an optimized version of such method because of <some cause> : if it's not ok I can happily change it into a runtime (fatal?) ex

Reply:
FYI now on PTO, will be able to perform code change when back on my desk by the end of month

Reply:
I can make the small changes to help this along if you agree with them. :)

Reply:
Many thanks @ijuma I appreciate and accept the help :)